apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -1566,8 +1573,11 @@ export async function render(
           ..._.mapValues(dynamicDependencies.coreElementScripts, (file) =>
             assets.coreElementAssetPath(file),
           ),
-          ..._.mapValues(dynamicDependencies.courseElementScripts, (file) =>
-            assets.courseElementAssetPath(course.commit_hash, locals.urlPrefix, file),
+          ..._.mapValues(
+            dynamicDependencies.courseElementScripts,
+            (file) =>
+              assets.courseElementAssetPath(course.commit_hash, locals.urlPrefix, file) +
+              (!idsEqual(question.course_id, course.id) ? `?variant_id=${variant.id}` : ''),
```

Comment:
Can we add some tests for this new behavior? Both checking whether we can access what we should, as well as can't access what we shouldn't.

Reply:
ya...there's currently no tests on these URLs at all, as far as I can tell, but I can add a few simple ones.

Reply:
I added tests for existing functionality that should have had tests as well as for new functionality.